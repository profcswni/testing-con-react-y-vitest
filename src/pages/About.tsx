import {Link} from "react-router-dom";

const About = () => {
    return (
        <div>
            <h1>About</h1>

            <p>
                This is a simple example of a Vite + React application.
            </p>

            <p>
                Estamos integrando React con Vite, vitest y react-testing-library.
            </p>

            <p>
                <Link to={'/'}>Back to home</Link>
            </p>
        </div>
    );
}

export default About;
