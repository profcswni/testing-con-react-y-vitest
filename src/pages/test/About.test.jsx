import { describe, it, expect, afterEach } from 'vitest';
import {cleanup, render, screen} from '@testing-library/react';
import About from "../About.tsx";
import {MemoryRouter} from "react-router-dom";

const RenderComponent = () => render(<MemoryRouter><About /></MemoryRouter>);

describe('About Page', () => {
    afterEach(async () => {
        cleanup()
    })

    it('componente renderizado correctamente', () => {
        const { asFragment } = RenderComponent();
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders the About heading', () => {
        const {getByText} = RenderComponent();
        expect(getByText('About')).toBeTruthy();
    });

    it('renders the description in English', () => {
        const {getByText} = RenderComponent()
        expect(getByText('This is a simple example of a Vite + React application.')).toBeTruthy();
    });

    it('renders the description in Spanish', () => {
        const {getByText} = RenderComponent()
        expect(getByText('Estamos integrando React con Vite, vitest y react-testing-library.')).toBeTruthy();
    });

    it('renders the Back to home link', () => {
        const {getByText} = RenderComponent()
        expect(getByText('Back to home')).toBeInTheDocument();
    });

    it('Back to home link has correct href', () => {
        const {getByText} = RenderComponent()
        expect(getByText('Back to home').closest('a')).toHaveAttribute('href', '/');
    });

    // Click en link to home
    it('Back to home link redirects to home page', () => {
        RenderComponent()
        const link = screen.getByText('Back to home');
        link.click();
        expect(screen.getByText(/About/i)).toBeInTheDocument();
    });
});
