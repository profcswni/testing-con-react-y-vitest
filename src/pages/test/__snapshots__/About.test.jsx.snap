// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About Page > componente renderizado correctamente 1`] = `
<DocumentFragment>
  <div>
    <h1>
      About
    </h1>
    <p>
      This is a simple example of a Vite + React application.
    </p>
    <p>
      Estamos integrando React con Vite, vitest y react-testing-library.
    </p>
    <p>
      <a
        data-discover="true"
        href="/"
      >
        Back to home
      </a>
    </p>
  </div>
</DocumentFragment>
`;
